spark/interpreter/src/main/java/org/apache/zeppelin/spark/IPySparkInterpreter.java

```diff
@@ -98,6 +99,16 @@ public BaseZeppelinContext buildZeppelinContext() {
     return sparkInterpreter.getZeppelinContext();
   }
 
+  @Override
+  public InterpreterResult interpret(String st, InterpreterContext context) {
+    InterpreterContext.set(context);
+    sparkInterpreter.populateSparkWebUrl(context);
+    String jobGroupId = Utils.buildJobGroupId(context);
+    String jobDesc = "Started by: " + Utils.getUserName(context.getAuthenticationInfo());
+    String setJobGroupStmt = "sc.setJobGroup('" +  jobGroupId + "', '" + jobDesc + "')";
+    return super.interpret(setJobGroupStmt +"\n" + st, context);
```

Comment:
@zjffdu This brought in a minor side effect described in https://issues.apache.org/jira/browse/ZEPPELIN-3484